src/game/boe.main.cpp

```diff
@@ -343,16 +343,16 @@ void init_boe(int argc, char* argv[]) {
 		Element& srand_element = pop_next_action("srand");
 		
 		std::string ts(srand_element.GetText());
-		srand(atoi(ts.c_str()));
+		game_rand.seed(atoi(ts.c_str()));
```

Comment:
This isn't really related to the PR, but I think `std::stoi` might be better than `atoi` here. Also removes the need to call `c_str()`.

Reply:
I can do a PR after this one that replaces atoi throughout the codebase, here are all the places we're using it:

```
$ rg atoi
src/location.cpp
245:	l.x = atoi(sstr.str().c_str());
249:	l.y = atoi(sstr.str().c_str());

src/fileio/xml-parser/tinyxml.cpp
578:			*i = atoi( s );
595:			*i = atoi( s->c_str() );
1275:	return atoi (value.c_str ());

src/fileio/xml-parser/tutorial_ticpp.txt
113:int attr = atoi( pszAttr );

src/dialogxml/dialogs/dialog.cpp
544:				eKeyMod mods = static_cast<eKeyMod>(atoi(info["mods"].c_str()));

src/game/boe.main.cpp
267:		eStartButton btn = static_cast<eStartButton>(atoi(next_action.GetText().c_str()));
346:		srand(atoi(ts.c_str()));
```

the xml-parser uses will stay the same. Everywhere else we call atoi() it is a clunky c_str() round-trip conversion like you pointed out

Reply:
xml-parser is an external library, so yeah, leaving it untouched makes sense. Looks like there aren't that many uses beside the xml-parser. It's likely they were added before 2011 so `stoi` wasn't even available.